### PR TITLE
[SPARK-25423][SQL] Output "dataFilters" in DataSourceScanExec.metadata

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

Output `dataFilters` in `DataSourceScanExec.metadata`.

## How was this patch tested?

unit tests

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/DataSourceScanExecRedactionSuite.scala [hunk 1]
```diff
@@ -83,4 +83,20 @@ class DataSourceScanExecRedactionSuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("FileSourceScanExec metadata") {
+    withTempDir { dir =>
+      val basePath = dir.getCanonicalPath
+      spark.range(0, 10).toDF("a").write.parquet(new Path(basePath, "foo=1").toString)
+      val df = spark.read.parquet(basePath).filter("a = 1")
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/DataSourceScanExecRedactionSuite.scala]: Hi, @wangyum . I know that you follow the style of the other test cases in this suite, but could you simplify like the following? We had better keep a single test case as simple as possible by excluding irrelevant stuffs.
```scala
    withTempPath { path =>
      val dir = path.getCanonicalPath
      spark.range(0, 10).toDF("a").write.parquet(dir)
      val df = spark.read.parquet(dir).filter("a = 1")
```

Comment (wangyum) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/DataSourceScanExecRedactionSuite.scala]: Thanks @dongjoon-hyun I fixed it.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Thank you, @wangyum !

**Pull request closed without merging**